<div style="max-width: 700px; margin: 20px auto 30px; padding: 20px; text-align: center; border-bottom: 2px solid #e5e7eb;">
  <a href="../session2/2_1_arma_models.ipynb" style="display: inline-block; padding: 10px 24px; background: linear-gradient(180deg, #38549c 0%, #14276c 100%); color: white; text-decoration: none; border-radius: 8px; font-weight: 600; font-size: 14px; box-shadow: 0 4px 12px rgba(56,84,156,0.3);">
    Part 2 - ARMA Model Estimation and Selection →
  </a>
</div>


<div style="max-width: 880px; margin: 20px auto 22px; padding: 0px; border-radius: 18px; border: 1px solid #e5e7eb; background: linear-gradient(180deg, #ffffff 0%, #f9fafb 100%); box-shadow: 0 8px 26px rgba(0,0,0,0.06); overflow: hidden;">

  <!-- Banner Header -->
  <div style="padding: 34px 32px 14px; text-align: center; line-height: 1.38;">
    <div style="font-size: 13px; letter-spacing: 0.14em; text-transform: uppercase; color: #6b7280; font-weight: bold; margin-bottom: 5px;">
      Session #1
    </div>
    <div style="font-size: 29px; font-weight: 800; color: #14276c; margin-bottom: 4px;">
      Univariate Time Series Model Simulations
    </div>
    <div style="font-size: 16.5px; color: #374151; font-style: italic; margin-bottom: 0;">
      Advanced Training School: Methods for Time Series
    </div>
  </div>

  <!-- Logo Section -->
  <div style="background: none; text-align: center; margin: 30px 0 10px;">
    <img src="https://www.cemfi.es/images/Logo-Azul.png" alt="CEMFI Logo" style="width: 158px; filter: drop-shadow(0 2px 12px rgba(56,84,156,0.05)); margin-bottom: 0;">
  </div>

  <!-- Name -->
  <div style="font-family: 'Times New Roman', Times, serif; color: #38549c; text-align: center; font-size: 1.22em; font-weight: bold; margin-bottom: 0px;">
    Jesus Villota Miranda © 2025
  </div>

  <!-- Contact info -->
  <div style="font-family: 'Times New Roman', Times, serif; color: #38549c; text-align: center; font-size: 1em; margin-top: 7px; margin-bottom: 20px;">
    <a href="mailto:jesus.villota@cemfi.edu.es" style="color: #38549c; text-decoration: none; margin-right:8px;" title="Email">
      jesus.villota@cemfi.edu.es
    </a>
    <span style="color:#9fa7bd;">|</span>
    <a href="https://www.linkedin.com/in/jesusvillotamiranda/" target="_blank" style="color: #38549c; text-decoration: none; margin-left:7px;" title="LinkedIn">
      LinkedIn
    </a>
  </div>
</div>


<div style="max-width: 700px; margin: 20px auto 30px; padding: 24px 32px; border-radius: 18px; border: 1px solid #e5e7eb; background: linear-gradient(180deg, #ffffff 0%, #f9fafb 100%); box-shadow: 0 8px 26px rgba(0,0,0,0.06);">

<div style="font-size: 22px; font-weight: 700; color: #14276c; margin-bottom: 16px; text-align: center;">
  Table of Contents
</div>

<div style="font-size: 15px; color: #374151; line-height: 1.8;">
  <div style="margin-bottom: 12px;"><strong>1.</strong> <a href="#section-1" style="color: #38549c; text-decoration: none;">White Noise Processes</a></div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">1.1 Uniform White Noise</div>
  <div style="margin-left: 20px; margin-bottom: 12px; font-size: 14px;">1.2 Gaussian White Noise</div>
  <div style="margin-bottom: 12px;"><strong>2.</strong> <a href="#section-2" style="color: #38549c; text-decoration: none;">MA(1) Models</a></div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">2.1 Invertible MA(1)</div>
  <div style="margin-left: 20px; margin-bottom: 12px; font-size: 14px;">2.2 Non-Invertible MA(1)</div>
  <div style="margin-bottom: 12px;"><strong>3.</strong> <a href="#section-3" style="color: #38549c; text-decoration: none;">AR(1) Models</a></div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">3.1 Stationary AR(1)</div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">3.2 Non-Stationary AR(1)</div>
  <div style="margin-left: 20px; margin-bottom: 12px; font-size: 14px;">3.3 Random Walk (Wiener Process)</div>
  <div style="margin-bottom: 12px;"><strong>4.</strong> <a href="#section-4" style="color: #38549c; text-decoration: none;">Markov Chains</a></div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">4.1 Binary Variable</div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">4.2 Binary Markov Chain</div>
  <div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">4.3 Unconditional Probability</div>
  <div style="margin-left: 20px; margin-bottom: 12px; font-size: 14px;">4.4 AR(1) Form of Markov Chain</div>
  <div style="margin-bottom: 0px;"><strong>5.</strong> <a href="#section-5" style="color: #38549c; text-decoration: none;">ARMA(1,1) Model</a></div>
</div>

</div>


<link rel="stylesheet" href="notebook_styles.css">


<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-1"></a>
<h2 class="styled-header">1) White Noise Processes</h2>

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Conceptual Foundation**: **White noise** is the fundamental building block of time series models. Think of it as the "pure randomness" that drives all stochastic processes—it has no pattern, no memory, and no predictability. Every complex time series model (AR, MA, ARMA) is constructed by transforming and combining white noise in different ways.

</div>

A white noise process $\{u_t\}$ is formally defined by three properties:

$$
\begin{aligned}
\mathbb{E}[u_t] &= 0 \quad \text{(zero mean)} \\
\mathbb{V}(u_t) &= \sigma^2 \quad \text{(constant variance)} \\
\mathbb{C}\text{ov}(u_t, u_s) &= 0 \quad \text{for } t \neq s \quad \text{(uncorrelated across time)}
\end{aligned}
$$

where:
- $\mathbb{E}[\cdot]$ denotes the expectation operator
- $\mathbb{V}(\cdot)$ denotes the variance
- $\mathbb{C}\text{ov}(\cdot, \cdot)$ denotes the covariance
- $\sigma^2$ is a constant variance parameter

**Key Properties**:

1. **No Autocorrelation**: The autocorrelation function satisfies $\rho(h) = 0$ for all $h \neq 0$, meaning observations at different time points are uncorrelated.

2. **No Memory**: Past values provide no information about future values. This is the Markov property in its simplest form—the process has no dependence structure.

3. **Foundation for Complex Models**: AR, MA, and ARMA models are all built by applying linear transformations to white noise, introducing structure through lagged relationships.

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Why This Matters**: White noise serves as the "innovation" or "shock" component in time series models. When we model economic or financial data, we're essentially decomposing the observed series into predictable components (trends, cycles) and unpredictable white noise (random shocks). Understanding white noise is essential because it represents the irreducible uncertainty in any stochastic process.

</div>


### 1.1 Uniform White Noise

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Objective**: Generate uniform white noise $e_1 \sim U(0,1)$ and demonstrate how nonlinear transformations affect the distribution. We compute $e_2 = e_1^2$ to show how squaring a uniform random variable changes its properties.

</div>

#### Mathematical Framework

We generate two processes:

$$
\begin{aligned}
e_1 &\sim U(0,1) \quad \text{(uniform distribution on [0,1])} \\
e_2 &= e_1^2 \quad \text{(square transformation)}
\end{aligned}
$$

**Theoretical Properties**:

- **$e_1$**: For $U \sim \text{Uniform}(0,1)$:
  - Mean: $\mathbb{E}[U] = \frac{1}{2} = 0.5$
  - Variance: $\mathbb{V}(U) = \frac{1}{12} \approx 0.0833$
  - Density: $f_U(u) = 1$ for $u \in [0,1]$, zero otherwise

- **$e_2$**: For $e_2 = U^2$ where $U \sim \text{Uniform}(0,1)$:
  - Mean: $\mathbb{E}[U^2] = \int_0^1 u^2 \cdot 1 \, du = \frac{1}{3} \approx 0.3333$
  - The distribution is no longer uniform—it's concentrated toward zero (since squaring small numbers yields smaller results)

<div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Important Note**: The transformation $e_2 = e_1^2$ demonstrates that **nonlinear transformations of random variables change their distribution**. While $e_1$ is uniformly distributed, $e_2$ follows a different distribution (specifically, $e_2$ has density $f_{e_2}(x) = \frac{1}{2\sqrt{x}}$ for $x \in [0,1]$).

</div>


In [3]:
* ============================================================================
* Simulation 1: Uniform White Noise
* ============================================================================
* Generate e1 = uniform() and e2 = e1 * e1

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate uniform random variable e1 on [0,1]
generate e1 = runiform()  // Generate uniform random numbers on [0,1]

* Generate e2 = e1 * e1 (square of e1)
generate e2 = e1 * e1  // Compute square of e1

* Display first 10 observations
list e1 e2 in 1/10  // Display first 10 observations of e1 and e2

* Summary statistics
display "Summary Statistics for Uniform White Noise:"
summarize e1 e2, detail  // Display detailed summary statistics for e1 and e2

* Generate time index for plotting
generate t = _n  // Create time index variable

* Time series plots
twoway (line e1 t), ///
    title("Uniform White Noise: e1 ~ U(0,1)") ///
    ytitle("e1") xtitle("Time") ///
    /* name(ts_uniform_e1, replace) // Save graph */

twoway (line e2 t), ///
    title("Squared Uniform: e2 = e1²") ///
    ytitle("e2") xtitle("Time") ///
    /* name(ts_uniform_e2, replace) // Save graph */

* Histogram plots
histogram e1, ///
    title("Histogram: Uniform White Noise e1") ///
    xtitle("e1") ytitle("Density") ///
    normal // Overlay normal curve for comparison ///
    /* name(hist_uniform_e1, replace) // Save graph */

histogram e2, ///
    title("Histogram: Squared Uniform e2 = e1²") ///
    xtitle("e2") ytitle("Density") ///
    /* name(hist_uniform_e2, replace) // Save graph */

* Theoretical mean of e1: E[U] = 0.5 for U ~ Uniform(0,1)
* Theoretical mean of e2: E[U^2] = 1/3 for U ~ Uniform(0,1)
display ""
display "Theoretical mean of e1 (Uniform(0,1)): 0.5"
display "Theoretical mean of e2 (E[U^2]): 1/3 ≈ 0.3333"





Number of observations (_N) was 0, now 1,000.




     +---------------------+
     |       e1         e2 |
     |---------------------|
  1. | .3576297    .127899 |
  2. | .4004426   .1603543 |
  3. | .6893833   .4752494 |
  4. | .5597356   .3133039 |
  5. | .5744513   .3299943 |
     |---------------------|
  6. | .2076905   .0431354 |
  7. | .0286627   .0008216 |
  8. | .6889245   .4746169 |
  9. | .4693434   .2202832 |
 10. | .2071526   .0429122 |
     +---------------------+

Summary Statistics for Uniform White Noise:


                             e1
-------------------------------------------------------------
      Percentiles      Smallest
 1%     .0086973       .0009672
 5%     .0499367       .0034476
10%      .094242       .0038868       Obs               1,000
25%     .2389324       .0039323       Sum of wgt.       1,000

50%     .4856097                      Mean           .4925308
                        Largest       Std. dev.      .2894846
75%      .737688       .99

#### Interpretation of Results

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 What the Analysis Reveals:**

1. **Uniform Distribution Properties**: The summary statistics for `e1` show a mean close to 0.5 and variance close to 0.0833, matching the theoretical values for $U(0,1)$. The histogram should display a flat, uniform distribution.

2. **Transformation Effects**: The squared variable `e2` has a mean close to $\frac{1}{3} \approx 0.3333$, confirming the theoretical expectation. The distribution of `e2` is **skewed toward zero** because squaring values in $[0,1]$ compresses the range—smaller numbers become even smaller when squared.

3. **Variance Comparison**: Notice that $\mathbb{V}(e_2) > \mathbb{V}(e_1)$ in the empirical results. This occurs because the transformation $e_2 = e_1^2$ creates a non-uniform distribution with different spread characteristics.

4. **Visual Patterns**: The time series plots show that both `e1` and `e2` exhibit no autocorrelation (random fluctuations), but `e2` has a different amplitude distribution due to the squaring transformation.

</div>

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Key Takeaway**: This simulation demonstrates that **nonlinear transformations preserve the white noise property (no autocorrelation) but change the marginal distribution**. This is important when modeling time series where we apply transformations (e.g., logarithms, squares) to the data.

</div>


### 1.2 Gaussian White Noise

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Objective**: Generate standard normal white noise $u \sim N(0,1)$ to demonstrate the most commonly used white noise distribution in time series modeling. Gaussian white noise is preferred because of the central limit theorem and its mathematical tractability.

</div>

We generate a standard normal white noise process:

$$
u_t \sim N(0,1)
$$

where:
- Mean: $\mathbb{E}[u_t] = 0$
- Variance: $\mathbb{V}[u_t] = 1$
- Density: $f(u) = \frac{1}{\sqrt{2\pi}} e^{-\frac{u^2}{2}}$

**Properties of Standard Normal White Noise**:

1. **Symmetry**: The distribution is symmetric around zero, with equal probability of positive and negative shocks.

2. **Bell-shaped Distribution**: Unlike uniform white noise, Gaussian white noise has a bell-shaped (normal) density curve, with most values concentrated near zero and fewer extreme values.

3. **Autocorrelation Check**: For true white noise, the autocorrelation function (ACF) should show $\rho(h) \approx 0$ for all $h \neq 0$, with values falling within the confidence bands.

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Comparison with Uniform White Noise**: While uniform white noise has bounded support $[0,1]$, Gaussian white noise has unbounded support $(-\infty, \infty)$. This makes Gaussian white noise more suitable for modeling economic and financial data where shocks can be arbitrarily large (though with decreasing probability).

</div>


#### Interpretation of Results

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 What the Analysis Reveals:**

1. **Distribution Verification**: The summary statistics show a mean close to 0 and variance close to 1, confirming the standard normal distribution. The histogram should display the characteristic bell-shaped curve, and the overlay normal curve should match closely.

2. **Autocorrelation Structure**: The ACF plot should show autocorrelations near zero for all lags $h \geq 1$, with most values falling within the confidence bands (gray area). This confirms the white noise property—no serial correlation.

3. **Random Fluctuations**: The time series plot shows random fluctuations around zero with no discernible pattern, trend, or cyclical behavior. This is the hallmark of white noise.

4. **Kurtosis Check**: The kurtosis should be close to 3 (the theoretical value for a normal distribution). Values significantly different from 3 indicate deviations from normality.

</div>


In [5]:
* ============================================================================
* Simulation 2: Gaussian White Noise
* ============================================================================
* Generate u = normal() (standard normal: N(0,1))

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 100000  // Set number of observations to 1000

* Generate standard normal random variable u ~ N(0,1)
generate u = rnormal()  // Generate standard normal random numbers

* Display first 10 observations
list u in 1/10  // Display first 10 observations of u

* Summary statistics
display "Summary Statistics for Gaussian White Noise:"
summarize u, detail  // Display detailed summary statistics for u

* Generate time index for plotting
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Time series plot
twoway (line u t), ///
    title("Gaussian White Noise: u ~ N(0,1)") ///
    ytitle("u") xtitle("Time") ///

* Histogram plot
histogram u, ///
    title("Histogram: Gaussian White Noise") ///
    xtitle("u") ytitle("Density") ///
    normal // Overlay normal curve ///

* Check autocorrelation (should be approximately zero for white noise)
ac u, lags(20) name(acf_wn, replace)

display ""
display "For white noise, autocorrelations should be approximately zero"
display "and fall within the confidence bands (gray area)"





Number of observations (_N) was 0, now 100,000.



     +-----------+
     |         u |
     |-----------|
  1. |     .9592 |
  2. |  -.001805 |
  3. |   .544077 |
  4. |  .0016287 |
  5. |  .3576809 |
     |-----------|
  6. |  1.878868 |
  7. |  2.754746 |
  8. | -.6125968 |
  9. |  .1973079 |
 10. |  1.610224 |
     +-----------+

Summary Statistics for Gaussian White Noise:


                              u
-------------------------------------------------------------
      Percentiles      Smallest
 1%    -2.311598      -4.417811
 5%    -1.639204      -4.196032
10%    -1.276607      -4.195403       Obs             100,000
25%    -.6756516      -4.060144       Sum of wgt.     100,000

50%      .000131                      Mean           .0014333
                        Largest       Std. dev.      .9999293
75%     .6754712       3.931678
90%     1.275755       4.076436       Variance       .9998585
95%     1.647029       4.091723       Skewness       .0059883
99%      2.34491  

<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-2"></a>
<h2 class="styled-header">2) MA(1) Models</h2>

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Conceptual Foundation**: The **Moving Average of order 1** (MA(1)) process models time series where the current value depends on the current shock and the previous period's shock. This captures short-term "memory" effects—how past innovations influence the present. MA models are particularly useful for modeling processes where shocks have temporary, finite-duration impacts.

</div>

The MA(1) process is defined as:

$$
x_t = u_t + \theta u_{t-1}
$$

where:
- $u_t \sim WN$ is white noise (the innovation/shock)
- $\theta$ is the MA coefficient that determines the strength of the lag-1 shock's influence
- The process depends only on current and one-period-lagged shocks

**Key Properties**:

1. **Autocorrelation Structure**:
   - First-order autocorrelation: $\rho(1) = \frac{\theta}{1+\theta^2}$
   - Higher-order autocorrelations: $\rho(h) = 0$ for $h \geq 2$
   - This "cut-off" property is characteristic of MA processes—autocorrelation drops to zero after a finite number of lags

2. **Invertibility Condition**: For $|\theta| < 1$, the MA(1) process is **invertible**, meaning it can be represented as an infinite AR process:
   $$
   x_t = -\sum_{j=1}^{\infty} (-\theta)^j x_{t-j} + u_t
   $$
   This allows us to express current shocks in terms of past observations.

3. **Non-Invertible Case**: When $|\theta| \geq 1$, the process is not invertible. The infinite AR representation does not converge, making it difficult to recover the underlying shocks from observed data.

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 16px; border-radius: 6px; margin: 16px 0;">

**⚠️ Warning**: Non-invertible MA processes have the same autocorrelation structure as invertible ones, creating an **identifiability problem**. For example, MA(1) with $\theta = 0.8$ and MA(1) with $\theta = 1.25$ produce identical autocorrelations. In practice, we always choose the invertible representation ($|\theta| < 1$) for estimation and forecasting.

</div>


#### Interpretation of Results

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 What the Analysis Reveals:**

1. **Autocorrelation Pattern**: The ACF for `x` should show a significant autocorrelation at lag 1 (approximately 0.4878) and near-zero autocorrelations for $h \geq 2$. This "cut-off" pattern is the signature of an MA(1) process.

2. **Variance Comparison**: The variance of `x` is larger than the variance of `u` because the MA(1) process combines current and lagged shocks: $\mathbb{V}(x_t) = (1+\theta^2)\sigma^2 = 1.64$ when $\sigma^2 = 1$.

3. **Process Comparison**: The time series plots comparing `x` and `y` show different patterns—`x` (with positive $\theta$) tends to have smoother transitions, while `y` (with negative $\theta$) shows more oscillatory behavior due to the negative correlation.

4. **Invertibility Verification**: Since $|\theta| = 0.8 < 1$, the process is invertible. This means we can successfully recover the underlying white noise shocks from the observed `x` values, which is essential for forecasting and model diagnostics.

</div>

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Key Takeaway**: The MA(1) process with $|\theta| < 1$ demonstrates the fundamental property of moving average models: **finite memory**. Shocks affect the process for exactly one period, then their influence disappears completely. This makes MA models ideal for capturing temporary, short-lived effects.

</div>


### 2.1 Invertible MA(1)

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Objective**: Simulate an invertible MA(1) process with $\theta = 0.8$ (satisfying $|\theta| < 1$) and demonstrate its properties. We also compute $y_t = u_t - \theta u_{t-1}$ to show how the sign of $\theta$ affects the process dynamics.

</div>

We simulate two processes:

$$
\begin{aligned}
x_t &= u_t + \theta u_{t-1} \quad \text{with } \theta = 0.8 \\
y_t &= u_t - \theta u_{t-1} \quad \text{(alternative with negative coefficient)}
\end{aligned}
$$

**Theoretical Moments**:

Assuming $u_t \sim N(0, \sigma^2)$ with $\sigma^2 = 1$:

**For $x_t = u_t + \theta u_{t-1}$**:

- **Expectation**: 
  $$
  \mathbb{E}[x_t] = \mathbb{E}[u_t + \theta u_{t-1}] = \mathbb{E}[u_t] + \theta \mathbb{E}[u_{t-1}] = 0 + \theta \cdot 0 = 0
  $$

- **Variance**:
  $$
  \mathbb{V}(x_t) = \mathbb{V}(u_t + \theta u_{t-1}) = \mathbb{V}(u_t) + \theta^2 \mathbb{V}(u_{t-1}) + 2\theta \mathbb{C}\text{ov}(u_t, u_{t-1})
  $$
  Since $u_t$ and $u_{t-1}$ are uncorrelated ($\mathbb{C}\text{ov}(u_t, u_{t-1}) = 0$):
  $$
  \mathbb{V}(x_t) = \sigma^2 + \theta^2 \sigma^2 = \sigma^2(1 + \theta^2) = 1 \cdot (1 + 0.64) = 1.64
  $$

- **Covariances**:
  $$
  \begin{aligned}
  \mathbb{C}\text{ov}(x_t, x_{t-1}) &= \mathbb{C}\text{ov}(u_t + \theta u_{t-1}, u_{t-1} + \theta u_{t-2}) \\
  &= \mathbb{C}\text{ov}(u_t, u_{t-1}) + \theta \mathbb{C}\text{ov}(u_t, u_{t-2}) + \theta \mathbb{C}\text{ov}(u_{t-1}, u_{t-1}) + \theta^2 \mathbb{C}\text{ov}(u_{t-1}, u_{t-2}) \\
  &= 0 + 0 + \theta \sigma^2 + 0 = \theta \sigma^2 = 0.8
  \end{aligned}
  $$
  
  For $h \geq 2$:
  $$
  \mathbb{C}\text{ov}(x_t, x_{t-h}) = 0 \quad \text{(no shared white noise terms)}
  $$

**For $y_t = u_t - \theta u_{t-1}$**:

- **Expectation**: 
  $$
  \mathbb{E}[y_t] = \mathbb{E}[u_t - \theta u_{t-1}] = 0
  $$

- **Variance**:
  $$
  \mathbb{V}(y_t) = \mathbb{V}(u_t - \theta u_{t-1}) = \sigma^2(1 + \theta^2) = 1.64 \quad \text{(same as $x_t$)}
  $$

- **Covariances**:
  $$
  \begin{aligned}
  \mathbb{C}\text{ov}(y_t, y_{t-1}) &= \mathbb{C}\text{ov}(u_t - \theta u_{t-1}, u_{t-1} - \theta u_{t-2}) \\
  &= \mathbb{C}\text{ov}(u_t, u_{t-1}) - \theta \mathbb{C}\text{ov}(u_t, u_{t-2}) - \theta \mathbb{C}\text{ov}(u_{t-1}, u_{t-1}) + \theta^2 \mathbb{C}\text{ov}(u_{t-1}, u_{t-2}) \\
  &= 0 - 0 - \theta \sigma^2 + 0 = -\theta \sigma^2 = -0.8
  \end{aligned}
  $$
  
  For $h \geq 2$:
  $$
  \mathbb{C}\text{ov}(y_t, y_{t-h}) = 0
  $$

- **Autocorrelations**

  - For $x_t$ with $\theta = 0.8$:
  $$
  \rho(1) = \frac{\mathbb{C}\text{ov}(x_t, x_{t-1})}{\sqrt{\mathbb{V}(x_t) \mathbb{V}(x_{t-1})}} = \frac{\theta \sigma^2}{\sigma^2(1+\theta^2)} = \frac{\theta}{1+\theta^2} = \frac{0.8}{1+0.64} = \frac{0.8}{1.64} \approx 0.4878
  $$

  - For $y_t$ with $\theta = 0.8$:
  $$
  \rho(1) = \frac{\mathbb{C}\text{ov}(y_t, y_{t-1})}{\sqrt{\mathbb{V}(y_t) \mathbb{V}(y_{t-1})}} = \frac{-\theta \sigma^2}{\sigma^2(1+\theta^2)} = \frac{-\theta}{1+\theta^2} = \frac{-0.8}{1.64} \approx -0.4878
  $$

**Where Does Autocorrelation Come From?**

The autocorrelation in MA(1) processes arises from the **shared white noise term** between consecutive observations:

- For $x_t = u_t + \theta u_{t-1}$ and $x_{t-1} = u_{t-1} + \theta u_{t-2}$:
  - Both depend on $u_{t-1}$, creating a **positive correlation** when $\theta > 0$
  - The covariance $\mathbb{C}\text{ov}(x_t, x_{t-1}) = \theta \sigma^2$ comes from the term $\theta \mathbb{C}\text{ov}(u_{t-1}, u_{t-1}) = \theta \sigma^2$
  
- For $y_t = u_t - \theta u_{t-1}$ and $y_{t-1} = u_{t-1} - \theta u_{t-2}$:
  - Both depend on $u_{t-1}$, but with **opposite signs**, creating a **negative correlation**
  - The covariance $\mathbb{C}\text{ov}(y_t, y_{t-1}) = -\theta \sigma^2$ comes from the term $-\theta \mathbb{C}\text{ov}(u_{t-1}, u_{t-1}) = -\theta \sigma^2$

- For $h \geq 2$, there are **no shared white noise terms** between $x_t$ and $x_{t-h}$ (or $y_t$ and $y_{t-h}$), so the autocorrelation is zero. This is the characteristic "cut-off" property of MA processes.

<!-- **Why Invertibility Matters**:

Since $|\theta| = 0.8 < 1$, the process is invertible. This means:
- We can recover $u_t$ from observed $x_t$ values
- The process has a convergent infinite AR representation
- Maximum likelihood estimation is well-behaved -->

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Key Takeaway**: The comparison between $x_t$ (positive $\theta$) and $y_t$ (negative $\theta$) illustrates how the sign of the MA coefficient affects the correlation structure. Positive $\theta$ creates positive autocorrelation, while negative $\theta$ creates negative autocorrelation at lag 1.

</div>


In [6]:
* ============================================================================
* Simulation 3: MA(1) - Invertible
* ============================================================================
* Simulate x_t = u_t + theta*u_{t-1} with theta=0.8, u_t ~ N(0,1)
* Then compute y_t = u_t - theta*u_{t-1}

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameter
scalar theta = 0.8  // Set MA(1) coefficient to 0.8

* Generate white noise u_t ~ N(0,1)
generate u = rnormal()  // Generate standard normal white noise

* Generate lagged white noise
generate u_lag1 = L.u  // Create lag-1 of u (u_{t-1})

* Simulate MA(1) process: x_t = u_t + theta*u_{t-1}
generate x = u + theta * u_lag1  // Compute MA(1) process

* Compute y_t = u_t - theta*u_{t-1}
generate y = u - theta * u_lag1  // Compute y_t transformation

* Display first 10 observations
list t u u_lag1 x y in 1/10  // Display first 10 observations

* Summary statistics
display "Summary Statistics for MA(1) Process:"
summarize u x y, detail  // Display detailed summary statistics

* Time series plots
twoway (line x t), ///
    title("MA(1) Process: x_t = u_t + 0.8*u_{t-1}") ///
    ytitle("x_t") xtitle("Time") ///
    name(ts_ma1_inv, replace) // Save graph

twoway (line y t), ///
    title("Transformed Process: y_t = u_t - 0.8*u_{t-1}") ///
    ytitle("y_t") xtitle("Time") ///
    name(ts_ma1_y, replace) // Save graph

* Autocorrelation function for x_t
ac x, lags(20)

* Autocorrelation function for y_t
ac y, lags(20)

display ""
display "For MA(1) with theta=0.8:"
display "Theoretical rho(1) = theta/(1+theta^2) = " %6.4f theta/(1+theta^2)
display "Theoretical rho(h) = 0 for h >= 2"





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit



(1 missing value generated)

(1 missing value generated)

(1 missing value generated)


     +----------------------------------------------------+
     |  t           u      u_lag1           x           y |
     |----------------------------------------------------|
  1. |  1       .9592           .           .           . |
  2. |  2    -.001805       .9592     .765555    -.769165 |
  3. |  3     .544077    -.001805     .542633    .5455211 |
  4. |  4    .0016287     .544077    .4368903   -.4336329 |
  5. |  5    .3576809    .0016287    .3589839     .356378 |
     |----------------------------------------------------|
  6. |  6    1.878868    .3576809    2.165012    1.592723 |
  7. |  7    2.754746    1.878868     4.25784    1.251652 |
  8. |  8   -.6125968    2.754746      1.5912   -2.816394 |
  9. |  9    .1973079   -.6125968   -.2927695    .6873854 |
 10. | 10    1.610224    .1

### 2.2 Non-Invertible MA(1)

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**⚠️ Warning - Identifiability Problem**: This simulation demonstrates a **non-invertible** MA(1) process where $1/\theta = 1.25 > 1$. While this process has the same autocorrelation structure as the invertible case, it cannot be represented as a convergent infinite AR process, creating estimation and forecasting difficulties.

</div>

We simulate a non-invertible MA(1) process:

$$
x_t = u_t + \frac{1}{\theta} u_{t-1} \quad \text{with } \theta = 0.8 \text{ (so } 1/\theta = 1.25 \text{)}
$$

**Key Issue**: Since $|1/\theta| = 1.25 > 1$, the process is **not invertible**.

**The Identifiability Problem**:

The MA(1) processes with coefficients $\theta$ and $1/\theta$ produce **identical autocorrelation functions**:

$$
\rho(1) = \frac{\theta}{1+\theta^2} = \frac{1/\theta}{1+(1/\theta)^2}
$$

For example:
- MA(1) with $\theta = 0.8$: $\rho(1) = 0.4878$
- MA(1) with $\theta = 1.25$: $\rho(1) = 0.4878$ (same!)

This means we cannot distinguish between these two processes based on autocorrelations alone.

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Why This Matters**: In practice, when estimating MA models, we **always choose the invertible representation** ($|\theta| < 1$) because:
1. It allows recovery of shocks from observations
2. It ensures stable forecasting
3. It makes maximum likelihood estimation well-behaved
4. It provides a unique parameterization

</div>


#### Interpretation of Results

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 What the Analysis Reveals:**

1. **Identical Autocorrelation**: The ACF for the non-invertible process shows the same pattern as the invertible case—significant autocorrelation at lag 1 (approximately 0.4878) and zero for $h \geq 2$. This confirms the identifiability problem.

2. **Increased Variance**: The variance of the non-invertible process is larger: $\mathbb{V}(x_t) = (1+(1/\theta)^2)\sigma^2 = (1+1.5625) = 2.5625$ when $\sigma^2 = 1$. This reflects the larger coefficient magnitude.

3. **Convergence Failure**: The non-invertible process cannot be written as a convergent infinite AR representation. Attempting to recover $u_t$ from $x_t$ would require an infinite sum that does not converge.

4. **Practical Implication**: When you see an MA(1) process with $|\theta| \geq 1$ in estimation output, it indicates a problem. Standard software packages automatically restrict estimation to the invertible region ($|\theta| < 1$) to ensure model identifiability and forecast stability.

</div>

<!-- <div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Important Note**: This simulation demonstrates why the invertibility condition is crucial in MA modeling. While both processes generate the same autocorrelation structure, only the invertible representation is useful for practical time series analysis, forecasting, and model diagnostics.

</div> -->


In [7]:
* ============================================================================
* Simulation 4: MA(1) - Not Invertible
* ============================================================================
* Simulate x_t = u_t + (1/theta)*u_{t-1} with theta=0.8
* Note: 1/theta = 1/0.8 = 1.25 > 1, so process is not invertible

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameter
scalar theta = 0.8  // Set theta parameter
scalar inv_theta = 1/theta  // Compute 1/theta = 1.25

* Generate white noise u_t ~ N(0,1)
generate u = rnormal()  // Generate standard normal white noise

* Generate lagged white noise
generate u_lag1 = L.u  // Create lag-1 of u (u_{t-1})

* Simulate non-invertible MA(1): x_t = u_t + (1/theta)*u_{t-1}
generate x = u + inv_theta * u_lag1  // Compute non-invertible MA(1) process

* Display first 10 observations
list t u u_lag1 x in 1/10  // Display first 10 observations

* Summary statistics
display "Summary Statistics for Non-Invertible MA(1) Process:"
summarize u x, detail  // Display detailed summary statistics

* Time series plot
twoway (line x t), ///
    title("Non-Invertible MA(1): x_t = u_t + 1.25*u_{t-1}") ///
    ytitle("x_t") xtitle("Time") ///
    name(ts_ma1_noninv, replace) // Save graph

* Autocorrelation function for x_t
ac x, lags(20) name(acf_ma1_noninv, replace)

display ""
display "For non-invertible MA(1) with 1/theta = " %6.4f inv_theta ":"
display "The process has the same autocorrelation structure as invertible MA(1)"
display "but cannot be represented as a convergent AR(infinity) process"





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit




(1 missing value generated)

(1 missing value generated)


     +----------------------------------------+
     |  t           u      u_lag1           x |
     |----------------------------------------|
  1. |  1       .9592           .           . |
  2. |  2    -.001805       .9592    1.197195 |
  3. |  3     .544077    -.001805    .5418208 |
  4. |  4    .0016287     .544077     .681725 |
  5. |  5    .3576809    .0016287    .3597168 |
     |----------------------------------------|
  6. |  6    1.878868    .3576809    2.325969 |
  7. |  7    2.754746    1.878868    5.103331 |
  8. |  8   -.6125968    2.754746    2.830836 |
  9. |  9    .1973079   -.6125968   -.5684381 |
 10. | 10    1.610224    .1973079    1.856859 |
     +----------------------------------------+

Summary Statistics for Non-Invertible MA(1) Process:


                              u
---------------------------

<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-3"></a>
<h2 class="styled-header">3) AR(1) Models</h2>

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Conceptual Foundation**: The **Autoregressive of order 1** (AR(1)) process models time series where the current value depends directly on the previous value. This captures **persistence** or **memory** in the process—how past values influence the present. AR models are fundamental for understanding mean reversion, trends, and long-term dependencies in economic and financial data.

</div>

The AR(1) process is defined as:

$$
x_t = \alpha x_{t-1} + u_t
$$

where:
- $u_t \sim WN$ is white noise (the innovation/shock)
- $\alpha$ is the AR coefficient that determines the strength of persistence
- The process depends on its own lagged value, creating serial correlation

**Key Properties**:

1. **Stationarity Condition**: For $|\alpha| < 1$, the process is **stationary**, meaning:
   - Mean: $\mathbb{E}[x_t] = 0$ (unconditional mean)
   - Variance: $\mathbb{V}(x_t) = \frac{\sigma^2}{1-\alpha^2}$ (finite and constant)
   - Autocorrelation: $\rho(h) = \alpha^h$ (exponential decay)

2. **Mean Reversion**: When $|\alpha| < 1$, the process exhibits **mean reversion**—deviations from the mean are temporary and the process tends to return to its long-run average.

3. **Unit Root Case**: When $\alpha = 1$, the process becomes a **random walk** (non-stationary):
   - Variance grows linearly with time: $\mathbb{V}(x_t) = t \sigma^2$
   - No mean reversion—shocks have permanent effects
   - First differences are white noise: $\Delta x_t = u_t$

4. **Explosive Case**: When $|\alpha| > 1$, the process is **explosive** (non-stationary):
   - Variance grows exponentially
   - Process diverges without bound

<div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Important Note**: The stationarity condition $|\alpha| < 1$ is crucial. When this condition is violated, standard statistical inference (hypothesis tests, confidence intervals) breaks down because the process has infinite variance or non-constant moments. Unit root testing is essential before applying AR models to real data.

</div>


### 3.1 Stationary AR(1)

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Objective**: Simulate a stationary AR(1) process with $\alpha = 0.8$ (satisfying $|\alpha| < 1$) and demonstrate mean reversion. We use an initial condition $x_1 = 100$ to show how the process gradually returns to its long-run mean of zero.

</div>

We simulate:

$$
x_t = \alpha x_{t-1} + u_t \quad \text{with } \alpha = 0.8, \quad u_t \sim N(0,1), \quad x_1 = 100
$$

**Theoretical Moments**:

Assuming $u_t \sim N(0, \sigma^2)$ with $\sigma^2 = 1$ and $|\alpha| < 1$ (stationarity condition):

- **Expectation**: 
  $$
  \mathbb{E}[x_t] = \mathbb{E}[\alpha x_{t-1} + u_t] = \alpha \mathbb{E}[x_{t-1}] + \mathbb{E}[u_t] = \alpha \mathbb{E}[x_{t-1}] + 0
  $$
  Under stationarity, $\mathbb{E}[x_t] = \mathbb{E}[x_{t-1}] = \mu$ (constant mean). Substituting:
  $$
  \mu = \alpha \mu + 0 \quad \Rightarrow \quad \mu(1-\alpha) = 0
  $$
  Since $|\alpha| < 1$, we have $1-\alpha \neq 0$, which implies:
  $$
  \mathbb{E}[x_t] = \mu = 0
  $$

- **Variance**:
  $$
  \mathbb{V}(x_t) = \mathbb{V}(\alpha x_{t-1} + u_t) = \alpha^2 \mathbb{V}(x_{t-1}) + \mathbb{V}(u_t) + 2\alpha \mathbb{C}\text{ov}(x_{t-1}, u_t)
  $$
  Since $u_t$ is white noise and independent of $x_{t-1}$ (which depends only on past shocks $u_{t-1}, u_{t-2}, \ldots$), we have $\mathbb{C}\text{ov}(x_{t-1}, u_t) = 0$. Under stationarity, $\mathbb{V}(x_t) = \mathbb{V}(x_{t-1}) = \gamma(0)$ (constant variance). Substituting:
  $$
  \gamma(0) = \alpha^2 \gamma(0) + \sigma^2 \quad \Rightarrow \quad \gamma(0)(1-\alpha^2) = \sigma^2
  $$
  Solving for $\gamma(0)$:
  $$
  \mathbb{V}(x_t) = \gamma(0) = \frac{\sigma^2}{1-\alpha^2} = \frac{1}{1-0.64} = \frac{1}{0.36} \approx 2.78
  $$

- **Covariances**:
  
  For $h = 1$:
  $$
  \begin{aligned}
  \gamma(1) = \mathbb{C}\text{ov}(x_t, x_{t-1}) &= \mathbb{C}\text{ov}(\alpha x_{t-1} + u_t, x_{t-1}) \\
  &= \alpha \mathbb{C}\text{ov}(x_{t-1}, x_{t-1}) + \mathbb{C}\text{ov}(u_t, x_{t-1}) \\
  &= \alpha \mathbb{V}(x_{t-1}) + 0 = \alpha \gamma(0)
  \end{aligned}
  $$
  
  For $h = 2$:
  $$
  \begin{aligned}
  \gamma(2) = \mathbb{C}\text{ov}(x_t, x_{t-2}) &= \mathbb{C}\text{ov}(\alpha x_{t-1} + u_t, x_{t-2}) \\
  &= \alpha \mathbb{C}\text{ov}(x_{t-1}, x_{t-2}) + \mathbb{C}\text{ov}(u_t, x_{t-2}) \\
  &= \alpha \gamma(1) + 0 = \alpha \cdot \alpha \gamma(0) = \alpha^2 \gamma(0)
  \end{aligned}
  $$
  
  By induction, for any $h \geq 0$:
  $$
  \gamma(h) = \alpha^h \gamma(0) = \alpha^h \frac{\sigma^2}{1-\alpha^2}
  $$
  
  For $\alpha = 0.8$ and $\sigma^2 = 1$:
  - $\gamma(0) = \frac{1}{1-0.64} = \frac{1}{0.36} \approx 2.78$
  - $\gamma(1) = 0.8 \cdot 2.78 \approx 2.22$
  - $\gamma(2) = 0.64 \cdot 2.78 \approx 1.78$
  - $\gamma(3) = 0.512 \cdot 2.78 \approx 1.42$


- **Autocorrelations**:

    The autocorrelation function is defined as:
    $$
    \rho(h) = \frac{\gamma(h)}{\gamma(0)} = \frac{\alpha^h \gamma(0)}{\gamma(0)} = \alpha^h
    $$

    For $\alpha = 0.8$:
    - $\rho(0) = 1$ (by definition)
    - $\rho(1) = 0.8$
    - $\rho(2) = 0.8^2 = 0.64$
    - $\rho(3) = 0.8^3 = 0.512$
    - $\rho(4) = 0.8^4 = 0.4096$

    This shows **exponential decay** in autocorrelation—the correlation between observations decreases geometrically as the lag increases.

**Where Does Autocorrelation Come From?**

The autocorrelation in AR(1) processes arises from the **recursive dependence** on past values:

- For $x_t = \alpha x_{t-1} + u_t$ and $x_{t-1} = \alpha x_{t-2} + u_{t-1}$:
  - $x_t$ **directly depends** on $x_{t-1}$ through the coefficient $\alpha$
  - $x_{t-1}$ in turn depends on $x_{t-2}$, which depends on $x_{t-3}$, creating a **chain of dependence**
  - This recursive structure means that $x_t$ depends on all past values $x_{t-1}, x_{t-2}, x_{t-3}, \ldots$, though with exponentially decaying weights
  
- The covariance $\mathbb{C}\text{ov}(x_t, x_{t-1}) = \alpha \gamma(0)$ comes from the direct linear relationship: when $x_{t-1}$ is high, $x_t$ tends to be $\alpha$ times as high (on average), creating positive correlation when $\alpha > 0$

- For $h \geq 1$, the autocovariance $\gamma(h) = \alpha^h \gamma(0)$ reflects the **infinite memory** of the AR(1) process—past values always matter, but their influence decays exponentially. This contrasts with MA(1), which has **finite memory** (autocorrelation is zero for $h \geq 2$)

- The exponential decay pattern $\rho(h) = \alpha^h$ means that:
  - When $|\alpha|$ is close to 1, the process has **strong persistence** (slow decay)
  - When $|\alpha|$ is close to 0, the process has **weak persistence** (fast decay toward white noise)

**Mean Reversion Dynamics**:

The process exhibits **mean reversion** because $|\alpha| = 0.8 < 1$. Starting from $x_1 = 100$, the process will gradually decay toward zero. 
From the Wold Decomposition of the AR(1), we have: 

$$
\mathbb{E}[x_t | x_1 = 100] = 100 \cdot \alpha^{t-1} = 100 \cdot 0.8^{t-1}
$$

This shows that the effect of the initial condition decays exponentially. 

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Key Takeaway**: The stationary AR(1) process demonstrates **persistence with decay**. While the process has memory (past values matter), this memory fades over time. The exponential decay in autocorrelation means that observations far apart are nearly uncorrelated, which is a hallmark of stationary processes.

</div>


In [8]:
* ============================================================================
* Simulation 5: AR(1) - Stationary
* ============================================================================
* Simulate x_t = alpha*x_{t-1} + u_t with alpha=0.8, u_t ~ N(0,1), x_1=100

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameter
scalar alpha = 0.8  // Set AR(1) coefficient to 0.8 (stationary)

* Generate white noise u_t ~ N(0,1)
generate u = rnormal()  // Generate standard normal white noise

* Initialize and simulate AR(1) process
generate x = .  // Initialize x as missing
replace x = 100 if t == 1  // Set initial condition x_1 = 100

* Recursively generate AR(1): x_t = alpha*x_{t-1} + u_t
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        replace x = alpha * L.x + u if t == `i'  // Compute AR(1) recursively
    }
}

* Display first 10 observations
list t u x in 1/10  // Display first 10 observations

* Summary statistics
display "Summary Statistics for Stationary AR(1) Process:"
summarize u x, detail  // Display detailed summary statistics

* Time series plot
twoway (line x t), ///
    title("Stationary AR(1): x_t = 0.8*x_{t-1} + u_t") ///
    ytitle("x_t") xtitle("Time") ///
    /* name(ts_ar1_stat, replace) // Save graph */

* Plot showing mean reversion (with zero reference line)
twoway (line x t), ///
    title("Stationary AR(1): Mean Reversion to Zero") ///
    ytitle("x_t") xtitle("Time") ///
    yline(0, lcolor(red) lpattern(dash)) ///
    /* name(ts_ar1_meanrev, replace) // Save graph */

* Autocorrelation function
ac x, lags(20) name(acf_ar1_stat, replace)

* Partial autocorrelation function
pac x, lags(20) name(pacf_ar1_stat, replace)

display ""
display "For stationary AR(1) with alpha = " %6.4f alpha ":"
display "Theoretical rho(h) = alpha^h (exponential decay)"
display "Mean reverts to zero (after initial condition effect dissipates)"




Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit



(1,000 missing values generated)

(1 real change made)


. list t u x in 1/10  

     +---------------------------+
     |  t           u          x |
     |---------------------------|
  1. |  1       .9592        100 |
  2. |  2    -.001805   79.99819 |
  3. |  3     .544077   64.54263 |
  4. |  4    .0016287   51.63573 |
  5. |  5    .3576809   41.66627 |
     |---------------------------|
  6. |  6    1.878868   35.21188 |
  7. |  7    2.754746   30.92425 |
  8. |  8   -.6125968    24.1268 |
  9. |  9    .1973079   19.49875 |
 10. | 10    1.610224   17.20922 |
     +---------------------------+

. display "Summary Statistics for Stationary AR(1) Process:"
Summary Statistics for Stationary AR(1) Process:

. summarize u x, detail  

                              u
-------------------------------------------------------------
      Percentiles      Smallest
 1%    -2.421662      -3

### 3.2 Non-Stationary AR(1)

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**⚠️ Explosive Process**: This simulation demonstrates an **explosive** AR(1) process where $\alpha = 1.0001 > 1$. The process is non-stationary with variance that grows exponentially over time, causing the series to diverge without bound.

</div>

We simulate:

$$
x_t = \alpha x_{t-1} + u_t \quad \text{with } \alpha = 1.0001 > 1, \quad u_t \sim N(0,1)
$$

**Why This is Problematic**:

When $|\alpha| > 1$:
- **Variance Explosion**: $\mathbb{V}(x_t)$ grows exponentially with time
- **No Mean Reversion**: The process does not return to a stable mean
- **Divergence**: Values become arbitrarily large (or small) as $t \to \infty$
- **Statistical Inference Fails**: Standard tests and confidence intervals are invalid

**Comparison with Stationary Case**:

- **Stationary** ($|\alpha| < 1$): Variance is bounded, process reverts to mean
- **Unit Root** ($\alpha = 1$): Variance grows linearly, shocks are permanent
- **Explosive** ($|\alpha| > 1$): Variance grows exponentially, process diverges

In [ ]:
* ============================================================================
* Simulation 6: AR(1) - Not Stationary (Explosive)
* ============================================================================
* Simulate x_t = alpha*x_{t-1} + u_t with alpha=1.0001, u_t ~ N(0,1)

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameter
scalar alpha = 1.0001  // Set AR(1) coefficient to 1.0001 (explosive, non-stationary)

* Generate white noise u_t ~ N(0,1)
generate u = rnormal()  // Generate standard normal white noise

* Initialize and simulate AR(1) process
generate x = .  // Initialize x as missing
replace x = 100 if t == 1  // Set initial condition x_1 = 0

* Recursively generate AR(1): x_t = alpha*x_{t-1} + u_t
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        replace x = alpha * L.x + u if t == `i'  // Compute AR(1) recursively
    }
}

* Display first 10 and last 10 observations
list t u x in 1/10  // Display first 10 observations
list t u x in -10/-1  // Display last 10 observations

* Summary statistics
display "Summary Statistics for Non-Stationary AR(1) Process:"
summarize u x, detail  // Display detailed summary statistics

* Time series plot
twoway (line x t), ///
    title("Non-Stationary AR(1): x_t = 1.0001*x_{t-1} + u_t (Explosive)") ///
    ytitle("x_t") xtitle("Time") ///
    name(ts_ar1_explosive, replace) // Save graph

* Note: Process explodes, so values become very large

display ""
display "For non-stationary AR(1) with alpha = " %6.4f alpha " > 1:"
display "The process is explosive - variance grows without bound"
display "No mean reversion - process diverges"





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit



(1,000 missing values generated)

(1 real change made)


. list t u x in 1/10  

     +---------------------------+
     |  t           u          x |
     |---------------------------|
  1. |  1       .9592        100 |
  2. |  2    -.001805   149.9982 |
  3. |  3     .544077   225.5414 |
  4. |  4    .0016287   338.3137 |
  5. |  5    .3576809   507.8282 |
     |---------------------------|
  6. |  6    1.878868   763.6212 |
  7. |  7    2.754746   1148.187 |
  8. |  8   -.6125968   1721.667 |
  9. |  9    .1973079   2582.698 |
 10. | 10    1.610224   3875.657 |
     +---------------------------+

. list t u x in -10/-1  

      +----------------------+
      |    t           u   x |
      |----------------------|
 991. |  991   -.2320457   . |
 992. |  992    .9659988   . |
 993. |  993    .0044365   . |
 994. |  994    1.065554   . |
 995. |  995   -.0965249   . |
      |-------

### 3.3 Random Walk (Wiener Process)

<div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Unit Root Process**: This simulation demonstrates a **random walk** (also called a **unit root process**) where $\alpha = 1$. This is the boundary case between stationarity and explosiveness, and it has profound implications for economic modeling and statistical inference.

</div>

We simulate:

$$
x_t = x_{t-1} + u_t \quad \text{with } u_t \sim N(0,1)
$$

This can be written as:

$$
x_t = x_0 + \sum_{j=1}^{t} u_j
$$

**Key Properties**:

1. **Variance Growth**: $\mathbb{V}[x_t] = t \sigma^2$ (variance grows **linearly** with time, not exponentially)

2. **Permanent Shocks**: Shocks have **permanent effects**—they never decay. A shock at time $s$ affects all future values $x_t$ for $t > s$.

3. **No Mean Reversion**: The process has no tendency to return to any fixed level. It "wanders" without bound.

4. **First Differences are White Noise**: $\Delta x_t = x_t - x_{t-1} = u_t$ (stationary!)

**Economic Interpretation**:

Random walks are commonly used to model:
- **Stock prices** (efficient market hypothesis)
- **Exchange rates** (uncovered interest parity)
- **GDP levels** (stochastic trends)

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Key Takeaway**: The random walk demonstrates that **first differencing** can transform a non-stationary process into a stationary one. This is the foundation of cointegration analysis and error correction models. If $\Delta x_t$ is white noise, then $x_t$ is a random walk, and standard inference must be applied to the differenced series.

</div>


In [12]:
* ============================================================================
* Simulation 7: AR(1) - Random Walk (Wiener Process)
* ============================================================================
* Simulate x_t = alpha*x_{t-1} + u_t with alpha=1, u_t ~ N(0,1)
* This is a random walk: x_t = x_{t-1} + u_t

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameter
scalar alpha = 1.0  // Set AR(1) coefficient to 1.0 (unit root)

* Generate white noise u_t ~ N(0,1)
generate u = rnormal()  // Generate standard normal white noise

* Initialize and simulate random walk: x_t = x_{t-1} + u_t
generate x = .  // Initialize x as missing
replace x = 0 if t == 1  // Set initial condition x_1 = 0

* Recursively generate random walk
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        replace x = L.x + u if t == `i'  // Compute random walk: x_t = x_{t-1} + u_t
    }
}

* Alternative: Use cumulative sum (more efficient for random walk)
* generate x = sum(u)  // Cumulative sum of u gives random walk

* Display first 10 and last 10 observations
list t u x in 1/10  // Display first 10 observations
list t u x in -10/-1  // Display last 10 observations

* Summary statistics
display "Summary Statistics for Random Walk:"
summarize u x, detail  // Display detailed summary statistics

* Time series plot
twoway (line x t), ///
    title("Random Walk: x_t = x_{t-1} + u_t") ///
    ytitle("x_t") xtitle("Time") ///
    name(ts_rw, replace) // Save graph

* Plot first differences (should be white noise)
generate dx = D.x  // First difference: Delta x_t = x_t - x_{t-1}
twoway (line dx t), ///
    title("First Differences of Random Walk (White Noise)") ///
    ytitle("Delta x_t") xtitle("Time") ///
    name(ts_rw_diff, replace) // Save graph

* Autocorrelation function (should show high persistence)
ac x, lags(20) name(acf_rw, replace)

* Partial autocorrelation function
pac x, lags(20) name(pacf_rw, replace)

* Autocorrelation of first differences (should be white noise)
ac dx, lags(20) name(acf_rw_diff, replace)

display ""
display "For random walk with alpha = 1:"
display "Variance grows linearly with time: Var(x_t) = t*sigma^2"
display "No mean reversion - process has infinite memory"
display "First differences are white noise: Delta x_t = u_t"





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit



(1,000 missing values generated)

(1 real change made)


. list t u x in 1/10  

     +---------------------------+
     |  t           u          x |
     |---------------------------|
  1. |  1       .9592          0 |
  2. |  2    -.001805   -.001805 |
  3. |  3     .544077    .542272 |
  4. |  4    .0016287   .5439007 |
  5. |  5    .3576809   .9015816 |
     |---------------------------|
  6. |  6    1.878868   2.780449 |
  7. |  7    2.754746   5.535195 |
  8. |  8   -.6125968   4.922598 |
  9. |  9    .1973079   5.119906 |
 10. | 10    1.610224    6.73013 |
     +---------------------------+

. list t u x in -10/-1  

      +-----------------------------+
      |    t           u          x |
      |-----------------------------|
 991. |  991   -.2320457   14.20963 |
 992. |  992    .9659988   15.17563 |
 993. |  993    .0044365   15.18007 |
 994. |  994    1.065554   16.2456

<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-4"></a>
<h2 class="styled-header">4) Markov Chains</h2>

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Conceptual Foundation**: **Markov chains** are discrete-state stochastic processes where the probability of transitioning to a new state depends only on the current state (Markov property). This "memoryless" property makes them powerful tools for modeling state-dependent dynamics in economics, finance, and other fields.

</div>

**Binary Markov Chain**: A two-state process with transition probabilities:
- $\mathbb{P}(x_t=0|x_{t-1}=0) = p$
- $\mathbb{P}(x_t=1|x_{t-1}=1) = q$

**Key Properties**:
- **Unconditional probability**: Under stationarity, $\pi = \mathbb{P}(x_t=0) = \frac{1-q}{2-p-q}$
- **AR(1) representation**: Can be written as $x_t = 1-p + (p+q-1)x_{t-1} + w_t$


<div style="background: linear-gradient(135deg, #f3e8ff 0%, #e9d5ff 100%); border-left: 4px solid #9333ea; padding: 18px; border-radius: 8px; margin: 20px 0;">

**📐 Derivation of the unconditional probability**

Under stationarity, the unconditional probability $\pi = \mathbb{P}(x_t=0)$ is constant over time, meaning $\mathbb{P}(x_t=0) = \mathbb{P}(x_{t-1}=0) = \pi$.

Using the **law of total probability**, we can express $\pi$ in terms of the transition probabilities:

$$\pi = \mathbb{P}(x_t=0) = \mathbb{P}(x_t=0|x_{t-1}=0)\mathbb{P}(x_{t-1}=0) + \mathbb{P}(x_t=0|x_{t-1}=1)\mathbb{P}(x_{t-1}=1)$$

Substituting the transition probabilities:
- $\mathbb{P}(x_t=0|x_{t-1}=0) = p$
- $\mathbb{P}(x_t=0|x_{t-1}=1) = 1 - \mathbb{P}(x_t=1|x_{t-1}=1) = 1-q$
- $\mathbb{P}(x_{t-1}=0) = \pi$
- $\mathbb{P}(x_{t-1}=1) = 1-\pi$

We obtain:

$$\begin{aligned}
\pi &= p\pi + (1-q)(1-\pi) \\
\pi &= p\pi + (1-q) - (1-q)\pi \\
\pi &= p\pi + 1 - q - \pi + q\pi \\
\pi - p\pi + \pi - q\pi &= 1 - q \\
\pi(1 - p + 1 - q) &= 1 - q \\
\pi(2 - p - q) &= 1 - q
\end{aligned}$$

Solving for $\pi$:

$$\pi = \frac{1-q}{2-p-q}$$

</div>

### 4.1 Binary Variable

Generate $b_t = \mathbf{1}\{u_t<0.2\}$ with $u_t \sim U(0,1)$, creating a binary indicator variable.


In [8]:
* ============================================================================
* Simulation 8: Binary Variable
* ============================================================================
* Generate b_t = 1(u_t < 0.2) with u_t ~ U(0,1)

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate uniform random variable u_t ~ U(0,1)
generate u = runiform()  // Generate uniform random numbers on [0,1]

* Generate binary variable: b_t = 1 if u_t < 0.2, 0 otherwise
generate b = (u < 0.2)  // Create binary indicator (1 if condition true, 0 otherwise)

* Display first 20 observations
list u b in 1/20  // Display first 20 observations

* Summary statistics
display "Summary Statistics for Binary Variable:"
summarize u b, detail  // Display detailed summary statistics

* Frequency table
tabulate b  // Display frequency table for binary variable

* Generate time index for plotting
generate t = _n  // Create time index variable

* Time series plot of binary variable
twoway (scatter b t, msize(tiny)), ///
    title("Binary Variable: b_t = 1(u_t < 0.2)") ///
    ytitle("b_t") xtitle("Time") ///
    ylabel(0 1) ///
    name(ts_binary, replace) // Save graph

* Bar chart of frequencies
graph bar (count), over(b) ///
    title("Frequency Distribution of Binary Variable") ///
    ytitle("Count") ///
    name(bar_binary, replace) // Save graph

display ""
display "Expected proportion of b=1: P(u < 0.2) = 0.2"
display "Expected proportion of b=0: P(u >= 0.2) = 0.8"





Number of observations (_N) was 0, now 1,000.




     +--------------+
     |        u   b |
     |--------------|
  1. | .3576297   0 |
  2. | .4004426   0 |
  3. | .6893833   0 |
  4. | .5597356   0 |
  5. | .5744513   0 |
     |--------------|
  6. | .2076905   0 |
  7. | .0286627   1 |
  8. | .6889245   0 |
  9. | .4693434   0 |
 10. | .2071526   0 |
     |--------------|
 11. | .0039323   1 |
 12. | .0130297   1 |
 13. | .4204224   0 |
 14. | .6161914   0 |
 15. | .8948836   0 |
     |--------------|
 16. | .4106361   0 |
 17. | .2607687   0 |
 18. | .0267289   1 |
 19. | .1079424   1 |
 20. |  .366684   0 |
     +--------------+

Summary Statistics for Binary Variable:


                              u
-------------------------------------------------------------
      Percentiles      Smallest
 1%     .0086973       .0009672
 5%     .0499367       .0034476
10%      .094242       .0038868       Obs               1,000
25%     .2389324       .0039323       Sum of wgt.       1,

### 4.2 Binary Markov Chain

Simulate a binary Markov chain with $x_0=0$, $\mathbb{P}(x_t=0|x_{t-1}=0)=p=0.7$, and $\mathbb{P}(x_t=1|x_{t-1}=1)=q=0.8$, using the construction:

$$x_t = (1-x_{t-1})\mathbf{1}\{u_t>p\} + x_{t-1}\mathbf{1}\{u_t<q\}$$

where $u_t \sim U(0,1)$.

**Theoretical unconditional probabilities:**

- $\mathbb{P}(x_t=0) =\pi = \frac{1-q}{2-p-q} = \frac{1-0.8}{2-0.7-0.8} = 0.4$
- $\mathbb{P}(x_t=1) = 1-\pi = 0.6$

In [ ]:
* ============================================================================
* Simulation 9: Binary Markov Chain
* ============================================================================
* Simulate binary Markov chain with x_0=0, P(x_t=0|x_{t-1}=0)=p=0.7, P(x_t=1|x_{t-1}=1)=q=0.8
* Construction: x_t = (1-x_{t-1})*1(u_t>p) + x_{t-1}*1(u_t<q)

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameters
scalar p = 0.7  // P(x_t=0|x_{t-1}=0) = 0.7
scalar q = 0.8  // P(x_t=1|x_{t-1}=1) = 0.8

* Generate uniform random variable u_t ~ U(0,1)
generate u = runiform()  // Generate uniform random numbers on [0,1]

* Initialize and simulate binary Markov chain
generate x = .  // Initialize x as missing
replace x = 0 if t == 1  // Set initial condition x_0 = 0

* Recursively generate Markov chain: x_t = (1-x_{t-1})*1(u_t>p) + x_{t-1}*1(u_t<q)
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        replace x = (1-L.x)*(u > p) + L.x*(u < q) if t == `i'  // Compute Markov chain
    }
}

* Display first 30 observations
list t u x in 1/30  // Display first 30 observations

* Summary statistics
display "Summary Statistics for Binary Markov Chain:"
summarize u x, detail  // Display detailed summary statistics

* Frequency table
tabulate x  // Display frequency table for binary variable

* Transition matrix (empirical)
generate x_lag1 = L.x  // Create lag-1 of x
tabulate x_lag1 x, row  // Display transition matrix with row percentages

* Time series plot of binary Markov chain
twoway (scatter x t, msize(tiny)), ///
    title("Binary Markov Chain") ///
    ytitle("x_t") xtitle("Time") ///
    ylabel(0 1) ///
    name(ts_markov, replace) // Save graph

* Bar chart of state frequencies
graph bar (count), over(x) ///
    title("Frequency Distribution of Markov Chain States") ///
    ytitle("Count") ///
    name(bar_markov, replace) // Save graph

display ""
display "Transition probabilities:"
display "P(x_t=0|x_{t-1}=0) = p = " %6.4f p
display "P(x_t=1|x_{t-1}=1) = q = " %6.4f q





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit




(1,000 missing values generated)

(1 real change made)


. list t u x in 1/30  

     +-------------------+
     |  t          u   x |
     |-------------------|
  1. |  1   .3576297   0 |
  2. |  2   .4004426   0 |
  3. |  3   .6893833   0 |
  4. |  4   .5597356   0 |
  5. |  5   .5744513   0 |
     |-------------------|
  6. |  6   .2076905   0 |
  7. |  7   .0286627   0 |
  8. |  8   .6889245   0 |
  9. |  9   .4693434   0 |
 10. | 10   .2071526   0 |
     |-------------------|
 11. | 11   .0039323   0 |
 12. | 12   .0130297   0 |
 13. | 13   .4204224   0 |
 14. | 14   .6161914   0 |
 15. | 15   .8948836   1 |
     |-------------------|
 16. | 16   .4106361   1 |
 17. | 17   .2607687   1 |
 18. | 18   .0267289   1 |
 19. | 19   .1079424   1 |
 20. | 20    .366684   1 |
     |-------------------|
 21. | 21   .0038868   1 |
 22. | 22   .9400924   0 |
 23. | 23   .6912759   0 |
 24

### 4.4 AR(1) Form of Markov Chain

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Objective**: Simulate the Markov chain using the AR(1) representation $x_t = 1-p + (p+q-1)x_{t-1} + w_t$ and compare with the direct Markov chain simulation. This demonstrates how discrete-state processes can be represented in continuous AR form.

</div>

To derive the AR(1) representation, we start by computing the conditional expectation of $x_t$ given $x_{t-1}$:

$$\mathbb{E}[x_t|x_{t-1}] = \mathbb{P}(x_t=1|x_{t-1}) \cdot 1 + \mathbb{P}(x_t=0|x_{t-1}) \cdot 0 = \mathbb{P}(x_t=1|x_{t-1})$$

Now, we express this conditional expectation in terms of $x_{t-1}$:

- When $x_{t-1}=0$: $\mathbb{E}[x_t|x_{t-1}=0] = \mathbb{P}(x_t=1|x_{t-1}=0) = 1-p$
- When $x_{t-1}=1$: $\mathbb{E}[x_t|x_{t-1}=1] = \mathbb{P}(x_t=1|x_{t-1}=1) = q$

We can combine these two cases into a single linear expression:

$$\begin{aligned}
\mathbb{E}[x_t|x_{t-1}] &= (1-p)(1-x_{t-1}) + qx_{t-1} \\
&= (1-p) - (1-p)x_{t-1} + qx_{t-1} \\
&= 1-p + (p+q-1)x_{t-1}
\end{aligned}$$

Now, define the error term as the deviation from the conditional expectation:

$$w_t = x_t - \mathbb{E}[x_t|x_{t-1}] = x_t - [1-p + (p+q-1)x_{t-1}]$$

By construction, $w_t$ has mean zero conditional on $x_{t-1}$: $\mathbb{E}[w_t|x_{t-1}] = \mathbb{E}[x_t|x_{t-1}] - \mathbb{E}[x_t|x_{t-1}] = 0$. Rearranging the error term definition, we obtain the AR(1) representation:

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 18px; border-radius: 8px; margin: 20px 0;">

**AR(1) Representation**:

$$x_t = 1-p + (p+q-1)x_{t-1} + w_t$$

This shows that the binary Markov chain can be expressed as an AR(1) process with:
- **Intercept**: $\beta_0 = 1-p$
- **AR(1) coefficient**: $\beta_1 = p+q-1$
- **Error term**: $w_t$ with $\mathbb{E}[w_t|x_{t-1}] = 0$

</div>

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Martingale Difference Sequence**: 

A **martingale difference sequence** (MDS) is a sequence of random variables $\{w_t\}$ that satisfies:

$$\mathbb{E}[w_t | \mathcal{F}_{t-1}] = 0$$

where $\mathcal{F}_{t-1}$ represents the information set (sigma-algebra) available at time $t-1$. 

**Key properties**:
- **Conditional mean zero**: The expected value of $w_t$ given past information is zero, but this does **not** imply independence.
- **Dependence on past**: Unlike white noise, $w_t$ can depend on $x_{t-1}$ and other past information—it only needs to have zero conditional expectation.
- **Variance may vary**: The conditional variance $\mathbb{V}(w_t | \mathcal{F}_{t-1})$ can depend on past information (heteroskedasticity).

In our Markov chain case, $w_t = x_t - \mathbb{E}[x_t|x_{t-1}]$ is an MDS because:
- $\mathbb{E}[w_t | x_{t-1}] = 0$ (by construction)
- But $w_t$ clearly depends on $x_{t-1}$ (its distribution changes based on the previous state)
- The variance of $w_t$ depends on $x_{t-1}$: $\mathbb{V}(w_t | x_{t-1}=0) \neq \mathbb{V}(w_t | x_{t-1}=1)$

This is why $w_t$ is **not** independent white noise, but rather a martingale difference sequence.

</div>


In [11]:
* ============================================================================
* Simulation 11: Markov Chain - AR(1) Form
* ============================================================================
* Simulate x_t = 1-p + (p+q-1)*x_{t-1} + w_t and compare with Markov chain simulation

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameters
scalar p = 0.7  // P(x_t=0|x_{t-1}=0) = 0.7
scalar q = 0.8  // P(x_t=1|x_{t-1}=1) = 0.8

* AR(1) coefficients
scalar beta0 = 1 - p  // Intercept: 1 - p
scalar beta1 = p + q - 1  // AR(1) coefficient: p + q - 1

* Method 1: Direct Markov chain simulation
generate u1 = runiform()  // Generate uniform random numbers for method 1
generate x_markov = .  // Initialize Markov chain
replace x_markov = 0 if t == 1  // Set initial condition
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        replace x_markov = (1-L.x_markov)*(u1 > p) + L.x_markov*(u1 < q) if t == `i'
    }
}

* Method 2: AR(1) form simulation
* Note: w_t is a martingale difference sequence
generate u2 = runiform()  // Generate uniform random numbers for method 2
generate x_ar1 = .  // Initialize AR(1) form
replace x_ar1 = 0 if t == 1  // Set initial condition

* Generate w_t (martingale difference sequence)
* w_t has conditional mean zero and appropriate variance
generate w = .  // Initialize w
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        * w_t depends on x_{t-1} and u_t
        * For x_{t-1}=0: w_t = 1(u_t>p) - (1-p)
        * For x_{t-1}=1: w_t = 1(u_t<q) - q
        replace w = (1-L.x_ar1)*((u2 > p) - (1-p)) + L.x_ar1*((u2 < q) - q) if t == `i'
        replace x_ar1 = beta0 + beta1*L.x_ar1 + w if t == `i'
    }
}

* Display first 30 observations
list t x_markov x_ar1 w in 1/30  // Display first 30 observations

* Summary statistics
display "Summary Statistics - Comparison:"
summarize x_markov x_ar1 w, detail  // Display detailed summary statistics

* Time series plot of w_t (martingale difference)
twoway (line w t), ///
    title("Martingale Difference Sequence: w_t") ///
    ytitle("w_t") xtitle("Time") ///
    name(ts_markov_w, replace) // Save graph

display ""
display "Correlation between Markov chain and AR(1) form: " %6.4f r(rho)
display ""
display "AR(1) form: x_t = " %6.4f beta0 " + " %6.4f beta1 "*x_{t-1} + w_t"
display "where w_t is a martingale difference sequence"





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit






(1,000 missing values generated)

(1 real change made)


. generate u2 = runiform()  

. generate x_ar1 = .  
(1,000 missing values generated)

. replace x_ar1 = 0 if t == 1  
(1 real change made)

. generate w = .  
(1,000 missing values generated)

. quietly {

. list t x_markov x_ar1 w in 1/30  

     +---------------------------------+
     |  t   x_markov       x_ar1     w |
     |---------------------------------|
  1. |  1          0           0     . |
  2. |  2          0   -1.19e-08   -.3 |
  3. |  3          0   -1.79e-08   -.3 |
  4. |  4          0           1    .7 |
  5. |  5          0           1    .2 |
     |---------------------------------|
  6. |  6          0           1    .2 |
  7. |  7          0           1    .2 |
  8. |  8          0   -1.19e-08   -.8 |
  9. |  9          0           1    .7 |
 10. | 10          0           1    .2 |
     |-----------

<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-5"></a>
<h2 class="styled-header">5) ARMA(1,1) Model</h2>

The **ARMA(1,1)** (Autoregressive Moving Average) process combines both AR and MA components:

$$x_t = \alpha x_{t-1} + u_t + \theta u_{t-1}$$

where $u_t \sim N(0,1)$ is white noise.

**Key Properties**:
- **Stationarity**: Requires $|\alpha| < 1$ (same as AR(1))
- **Invertibility**: Requires $|\theta| < 1$ (same as MA(1))
- **Autocorrelation**: More complex pattern than pure AR or MA, with both persistence and shock effects
- **Parsimony**: Can capture autocorrelation patterns with fewer parameters than pure AR or MA models


### ARMA(1,1) Simulation

Simulate $x_t = \alpha x_{t-1} + u_t + \theta u_{t-1}$ with $\alpha=0.9$, $\theta=-0.8$, and $u_t \sim N(0,1)$.


In [2]:
* ============================================================================
* Simulation 12: ARMA(1,1)
* ============================================================================
* Simulate x_t = alpha*x_{t-1} + u_t + theta*u_{t-1} with alpha=0.9, theta=-0.8, u_t ~ N(0,1)

clear all  // Clear memory and remove all variables
set seed 12345  // Set random seed for reproducibility
set obs 1000  // Set number of observations to 1000

* Generate time index
generate t = _n  // Create time index variable
tsset t  // Declare time series structure

* Set parameters
scalar alpha = 0.9  // AR(1) coefficient
scalar theta = -0.8  // MA(1) coefficient

* Generate white noise u_t ~ N(0,1)
generate u = rnormal()  // Generate standard normal white noise

* Generate lagged white noise
generate u_lag1 = L.u  // Create lag-1 of u (u_{t-1})

* Initialize and simulate ARMA(1,1) process
generate x = .  // Initialize x as missing
replace x = 0 if t == 1  // Set initial condition x_1 = 0

* Recursively generate ARMA(1,1): x_t = alpha*x_{t-1} + u_t + theta*u_{t-1}
quietly {
    forvalues i = 2/1000 {  // Loop from observation 2 to 1000
        replace x = alpha * L.x + u + theta * u_lag1 if t == `i'  // Compute ARMA(1,1)
    }
}

* Display first 10 observations
list t u u_lag1 x in 1/10  // Display first 10 observations

* Summary statistics
display "Summary Statistics for ARMA(1,1) Process:"
summarize u x, detail  // Display detailed summary statistics

* Time series plot
twoway (line x t), ///
    title("ARMA(1,1): x_t = 0.9*x_{t-1} + u_t - 0.8*u_{t-1}") ///
    ytitle("x_t") xtitle("Time") ///
    name(ts_arma11, replace) // Save graph

* Autocorrelation function
ac x, lags(20) name(acf_arma11, replace)

* Partial autocorrelation function
pac x, lags(20) name(pacf_arma11, replace)

display ""
display "For ARMA(1,1) with alpha = " %6.4f alpha " and theta = " %6.4f theta ":"
display "Stationarity condition: |alpha| < 1 (satisfied: " %6.4f abs(alpha) " < 1)"
display "Invertibility condition: |theta| < 1 (satisfied: " %6.4f abs(theta) " < 1)"
display "The ACF shows exponential decay (from AR component)"
display "The PACF also shows exponential decay (from MA component)"





Number of observations (_N) was 0, now 1,000.



Time variable: t, 1 to 1000
        Delta: 1 unit




(1 missing value generated)

(1,000 missing values generated)

(1 real change made)


. list t u u_lag1 x in 1/10  

     +----------------------------------------+
     |  t           u      u_lag1           x |
     |----------------------------------------|
  1. |  1       .9592           .           0 |
  2. |  2    -.001805       .9592    -.769165 |
  3. |  3     .544077    -.001805   -.1467274 |
  4. |  4    .0016287     .544077   -.5656876 |
  5. |  5    .3576809    .0016287   -.1527409 |
     |----------------------------------------|
  6. |  6    1.878868    .3576809    1.455256 |
  7. |  7    2.754746    1.878868    2.561382 |
  8. |  8   -.6125968    2.754746   -.5111495 |
  9. |  9    .1973079   -.6125968    .2273508 |
 10. | 10    1.610224    .1973079    1.656993 |
     +----------------------------------------+

. display "Summary Statistics for ARMA(1,1) Process:"
Su

<div style="max-width: 700px; margin: 20px auto 30px; padding: 20px; text-align: center; border-bottom: 2px solid #e5e7eb;">
  <a href="../session2/2_1_arma_models.ipynb" style="display: inline-block; padding: 10px 24px; background: linear-gradient(180deg, #38549c 0%, #14276c 100%); color: white; text-decoration: none; border-radius: 8px; font-weight: 600; font-size: 14px; box-shadow: 0 4px 12px rgba(56,84,156,0.3);">
    Part 2 - ARMA Model Estimation and Selection →
  </a>
</div>
